rust/xaynet-server/src/storage/redis.rs

```diff
@@ -630,7 +630,10 @@ mod tests {
             .add_sum_participant(pk, epk)
             .await
             .unwrap();
-        assert_eq!(key_already_exist, AddSumParticipant::AlreadyExists);
+        assert!(matches!(
+            key_already_exist.into_inner().unwrap_err(),
+            SumDictAddError::AlreadyExists
+        ));
```

Comment:
i was wondering what's the reason for `matches!`? wouldn't
```rust
assert_eq!(
    key_already_exist.into_inner().unwrap_err(),
    SumDictAddError::AlreadyExists,
);
```
result in the same outcome?

Reply:
That's right, but I would have to add `#[derive(PartialEq)]` to the `SeedDictUpdateError` enum. With `matches!` it works without `PartialEq`.

Reply:
i see, fine for me to do it this way. another way would be to annotate the struct with `#[cfg_attr(test, derive(PartialEq))]`, then it will only derive if the code is compiled for tests.